### DISCUSSION TITLE
Issue with application.yml

### DISCUSSION THREAD

**Discussion by SebDeLille** [quarkusio/quarkus#23092] [Community]
Hello,

I have an issue with Quarkus and Kafka and Apicurio when i use an application.yml file. I have this exception: 
Caused by: javax.enterprise.inject.spi.DeploymentException: org.apache.kafka.common.KafkaException: java.lang.IllegalArgumentException: Missing registry base url, set apicurio.registry.url

When i convert it in an application.properties, my quarkus app run correctly. 

Is it a known bug? or do I have an error in my yml file?

Sébastien

[application.zip](https://github.com/quarkusio/quarkus/files/7913820/application.zip)

**Comment by ozangunalp** [comment]
@SebDeLille I realized that you are missing apicurio-registry-avro dependency in your `pom.xml`.
If you add the following it should work
```xml
<dependency>
    <groupId>io.quarkus</groupId>
    <artifactId>quarkus-apicurio-registry-avro</artifactId>
</dependency>
```

**Reply by ozangunalp** [reply]
You'll also need to change the package names of classes referenced inside your config file.

**Comment by SebDeLille** [comment]
@ozangunalp 
I don't quite understand. My project works fine with application.properties. Why should I change pom.xml when I replace application.properties with application.yml?

**Reply by ozangunalp** [reply]
I was just checking that. Actually, you are missing the `quarkus-config-yaml` dependency for using yaml config files: https://quarkus.io/guides/config-yaml

**Comment by SebDeLille** [comment]
oops, my bad. thanks for your help
